src/renderer/dx/DxFontInfo.cpp

```diff
@@ -202,6 +202,63 @@ void DxFontInfo::SetFromEngine(const std::wstring_view familyName,
     return face;
 }
 
+// Routine Description:
+// - Locates a suitable font face from the given information using more advanced match filtering on font sets only
+//   possible with Win10 and higher versions of DirectWrite
+// Arguments:
+// - fontCollection - The font collection to dig through
+// - localeName - The locale to search for appropriate fonts.
+// Return Value:
+// - Smart pointer holding interface reference for queryable font data.
+// - The pointer can be empty/null if there was a failure or we're below Win10 and need to fallback
+[[nodiscard]] Microsoft::WRL::ComPtr<IDWriteFontFace1> DxFontInfo::_TryWin10FontMatchingLogic(gsl::not_null<IDWriteFontCollection*> fontCollection,
+                                                                                              std::wstring& localeName)

```

Comment:
Don't we need to pass the `axes` too, for weight/width/slope/(or maybe even user defined ones)? Otherwise the individual faces within the filtered font subset from `GetFilteredFonts` won't be in any particular order. `IDWriteFontSet1::GetMatchingFonts` applied to the filtered font set will return them ordered in priority ordered with the closest match at index 0.

Reply:
Yeah that's acting a bit weird right now. A few reasons.

We were using the definition of axes with the weight/stretch/style enums... and now the newer APIs want those as numerical axes values... which I think we have in the other font-related class not this one.

We default to normal/normal/normal on those but if someone says in text form "Something Something Black".... I want "Black" to be more specific than the default normal... Though if they say "Something Something Black" --> WEIGHT_ENUM_VALUE = LIGHT.... I want the light one. And if they say "Something Something Black" --> WEIGHT_AXIS_VALUE = 342... I want it to try to resolve the nearest to 342 weight...

Reply:
"Arial Black" should still work, because DWRITE_FONT_PROPERTY_ID_FULL_NAME will match last, and there will only be one font in the resulting filtered set (unless it's a true variable font rather than discrete font file, in which case you do want the axes).

If I could rewrite history, I'd eliminate the old enums, but we couldn't see the future 10 years later 😅. Do you need a mapping between the old and new ones? I can write one in this comment...